unified-prototype/unified-plugin/build-logic/src/main/kotlin/build-logic.publishing.gradle.kts

```diff
@@ -0,0 +1,44 @@
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+plugins {
+    id("com.gradle.plugin-publish")
+    id("com.github.johnrengelman.shadow") apply false
+    signing
+}
+
+val commonOnlyJarDependencyScope = configurations.dependencyScope("commonOnlyJarDependencyScope")
+val commonOnlyJar = configurations.resolvable("commonOnlyJar") {
+    extendsFrom(commonOnlyJarDependencyScope.get())
+}
+
+dependencies {
+    commonOnlyJarDependencyScope(project(":plugin-common")) {
+        isTransitive = false
+    }
+}
+
+tasks.shadowJar {
+    configurations = listOf(commonOnlyJar.get())
+    archiveClassifier.set("")
+    relocate(
+        "org.gradle.api.experimental.common",
+        Matcher.quoteReplacement("org.gradle.api.experimental.common.0shaded.into.${project.name}"),
```

Comment:
```suggestion
        Matcher.quoteReplacement("org.gradle.api.experimental.common.shaded.into.${project.name}"),
```

Reply:
Are you sure it's a legal identifier for a package name?

The [JLS](https://docs.oracle.com/javase/specs/jls/se7/html/jls-6.html#jls-6.1) says it's not a valid package if:

>  If any of the resulting package name components start with a digit, or any other character that is not allowed as an initial character of an identifier, have an underscore prefixed to the component.

So I'm surprised this works.

Reply:
Right, the _Java_ spec does not allow this (that's the point). The JVM spec does, because packages have no restrictions: https://docs.oracle.com/javase/specs/jvms/se22/html/jvms-4.html#jvms-4.2.3

Reply:
Ah, yes.  Makes sense now.  :-)